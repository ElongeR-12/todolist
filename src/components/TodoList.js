import React from 'react'
import Todo from './Todo'

function TodoList({todos, setTodos}) {
    
  return(
      <>
        <h1>Todo List</h1>
        { 
            todos && todos.map((todo)=>(<Todo key={todo.id} todo={todo} setTodos={setTodos} todos={todos}/>))
        }
      </>
  )
  
}

export default TodoList