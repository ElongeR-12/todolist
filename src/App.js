
import './App.css';
import React from 'react'
import TodoList from './components/TodoList';
function App() {

 const [todos,setTodos]= React.useState( [
        {
          id:"fdsqfdkslqfdsq",
          state:false,
          description:"Lorem ipsum dolor sit amet",
          title:"title one"
        },
        {
          id:"fdqfdqslk",
          state:false,
          description:"Lorem ipsum dolo",
          title:"title two"
        },
        {
          id:"dfsqfdsqfdq",
          state:false,
          description:"Lorem ipsum",
          title:"title three"
        }
      ]
 )
    return (
      <div className="App">
      <TodoList todos={todos} setTodos={setTodos}/>
    </div>
    )
}

export default App
 
