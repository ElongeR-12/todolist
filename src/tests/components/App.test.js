import React from 'react';
import { shallow } from 'enzyme';
import TodoList from '../../components/TodoList';
import App from '../../App';

describe("App test validation",()=>{
  test('should test app has div parent element with className App', () => {
    const appWrapper = shallow(<App/>);
    const wrapper = appWrapper.find('div');
    expect(wrapper.hasClass('App')).toBeTruthy()
});
test('should test app contain TodoList components', () => {
  const appWrapper = shallow(<App/>);
  const wrapper = appWrapper.find(TodoList);
  expect(wrapper).toHaveLength(1)
});
test('should test props to todoList components', () => {
  const appWrapper = shallow(<App/>);
  const wrapper = appWrapper.find(TodoList);
  expect(wrapper).toHaveProp("todos")
});
test('has state',()=>{
  const appWrapper = shallow(<App/>);
  const appState = appWrapper.state();
  expect(appState).not.toBeNull();
})
test('state lenght great than 0',()=>{
  const appWrapper = shallow(<App/>);
  const appState = appWrapper.state();
  expect(appState.todos).toHaveLength(3);
})
test('todos props of todolists should be an array of todo, and contain initial todos', () => {
  const wrapper = shallow(<App/>);
  expect(wrapper).toMatchSnapshot();
});
  
})
