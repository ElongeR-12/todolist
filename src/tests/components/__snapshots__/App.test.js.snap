// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test validation todos props of todolists should be an array of todo, and contain initial todos 1`] = `
<div
  className="App"
>
  <TodoList
    setTodos={[Function]}
    todos={
      Array [
        Object {
          "description": "Lorem ipsum dolor sit amet",
          "id": "fdsqfdkslqfdsq",
          "state": false,
          "title": "title one",
        },
        Object {
          "description": "Lorem ipsum dolo",
          "id": "fdqfdqslk",
          "state": false,
          "title": "title two",
        },
        Object {
          "description": "Lorem ipsum",
          "id": "dfsqfdsqfdq",
          "state": false,
          "title": "title three",
        },
      ]
    }
  />
</div>
`;
